src/python/pants/backend/python/target_types_test.py

```diff
@@ -364,35 +366,90 @@ def test_parse_requirements_file() -> None:
     }
 
 
-def test_python_distribution_dependency_injection() -> None:
+def test_inject_python_distribution_dependencies() -> None:
     rule_runner = RuleRunner(
         rules=[
             inject_python_distribution_dependencies,
+            resolve_pex_entry_point,
+            resolve_python_distribution_entry_points,
+            *import_rules(),
+            *python_sources.rules(),
             QueryRule(InjectedDependencies, [InjectPythonDistributionDependencies]),
         ],
-        target_types=[PythonDistribution, PexBinary],
+        target_types=[PythonDistribution, PythonRequirementLibrary, PythonLibrary, PexBinary],
         objects={"setup_py": PythonArtifact},
     )
+    rule_runner.add_to_build_file(
+        "",
+        dedent(
+            """\
+            python_requirement_library(
+                name='ansicolors',
+                requirements=['ansicolors'],
+                module_mapping={'ansicolors': ['colors']},
```

Comment:
What I mean with testing colors is to add an entry point like this: `colors_ep = "colors"`, such that your dep inference rule should pick up a dep on the `:ansicolors` target.

Reply:
Done. https://github.com/pantsbuild/pants/pull/11872/files#diff-a2ccc3f505ec7e33785464002b3c6192b325d97fcdcda65ad11876f2b5d7a500R423-R431 :)